src/main/java/it/gov/pagopa/payhub/activities/service/FlowValidatorService.java

```diff
@@ -0,0 +1,33 @@
+package it.gov.pagopa.payhub.activities.service;
+
+import it.gov.digitpa.schemas._2011.pagamenti.CtFlussoRiversamento;
+import it.gov.pagopa.payhub.activities.dto.IngestionFlowFileDTO;
+import it.gov.pagopa.payhub.activities.exception.InvalidFlowDataException;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service class responsible for validating flow data, specifically ensuring that the organization
+ * details in the flow match the expected organization.
+ */
+@Lazy
+@Service
+public class FlowValidatorService {
```

Comment:
This is specialized on PaymentsReporting
```suggestion
public class PaymentsReportingIngestionFlowFileValidatorService {
```